### PR TITLE
Multi thread project

### PR DESCRIPTION
**Pull request by YuriyLykhva**

### CODE REVIEW

File: core/src/main/java/core/util/Constants.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package core.util;
+
+public class Constants {
+
+    public static final int WAIT_TIMEOUT_SECONDS = 10;
+    public static final String SIGNIN_PAGE_URL = "http://localhost:8080";
+    public static final boolean USE_REMOTE_WEB_DRIVER = true;
+    public static final String REMOTE_WEB_SERVER_URL = "http://192.168.0.101:4444";
```
Comment (Stasnazarenko) [inline core/src/main/java/core/util/Constants.java:8]: better to have at app.properties file,in order define as system var,and rewrite in maven launch once s required

Comment (Stasnazarenko) [reply core/src/main/java/core/util/Constants.java:8]: all of that

File: core/src/main/java/core/util/JsExecutor.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+public class JsExecutor {
+
+    protected void jsClickElement(WebElement element) {
+        JavascriptExecutor executor = (JavascriptExecutor) getDriver();
+        executor.executeScript("arguments[0].click();", element);
+    }
+
+    protected void jsViewElement(WebElement element) {
+        JavascriptExecutor executor = (JavascriptExecutor) getDriver();
```
Comment (Stasnazarenko) [inline core/src/main/java/core/util/JsExecutor.java:16]: why not to use a single thread-safe instance of the executor over this class?

File: core/src/main/java/core/util/TestListener.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import java.io.File;
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class TestListener implements ITestListener {
+
+    @Override
+    public void onTestSuccess(ITestResult iTestResult) {
+        log("Test '" + iTestResult.getName() + "' PASSED");
```
Comment (Stasnazarenko) [inline core/src/main/java/core/util/TestListener.java:20]: String format is preferable way of defining such constructions

File: core/src/main/java/core/util/TestListener.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                    ".//target/screenshots/"
+                            + getCurrentTimeAsString() +
+                            ".png"));
+        } catch (IOException e) {
+            log("Failed to save screenshot: " + e.getLocalizedMessage());
+        }
+    }
+
+    private void log(String methodName) {
+        System.out.println(methodName);
```
Comment (Stasnazarenko) [inline core/src/main/java/core/util/TestListener.java:62]: logging to user console,is a bit strange approach
use common logger for logging as system log

File: pom.xml [hunk 5]
```diff
@@ -33,93 +39,11 @@
                         <allure.link.tms.pattern>https://example.org/browse/{}</allure.link.tms.pattern>
                     </systemPropertyVariables>
                     <suiteXmlFiles>
-                        <suiteXmlFile>testng.xml</suiteXmlFile>
+<!--                        <suiteXmlFile>testng.xml</suiteXmlFile>-->
```
Comment (Stasnazarenko) [inline pom.xml:42]: commneted line

File: Jenkinsfile-2 [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+                echo 'Hello Stage 2'
+            }
+        }
+        stage('3') {
+            steps {
+                echo 'Hello on Stage 3'
+            }
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (YuriyLykhva) [reply Jenkinsfile-2]: done

File: pom.xml (hunk 5 above)
Comment (YuriyLykhva) [reply pom.xml:42]: done

File: core/src/main/java/core/util/Constants.java (hunk 1 above)
Comment (YuriyLykhva) [reply core/src/main/java/core/util/Constants.java:8]: done

File: core/src/main/java/core/util/TestListener.java (hunk 4 above)
Comment (YuriyLykhva) [reply core/src/main/java/core/util/TestListener.java:62]: done as well)

File: Jenkinsfile-2 (hunk 6 above)
[the remark this answers is not in the document]
Comment (Stasnazarenko) [reply Jenkinsfile-2]: +

**Pull request merged**